src/components/MessageList/utils/getReadStates.js

```diff
@@ -0,0 +1,32 @@
+export const getReadStates = (messages, read) => {
+  // create object with empty array for each message id
+  const readData = {};
+
+  for (const message of messages) {
+    // Filter out date seperators
+    if (!message || !message.id) {
+      continue;
+    }
+    readData[message.id] = [];
+  }
+
+  for (const readState of Object.values(read)) {
+    if (readState.last_read == null) {
```

Comment:
`==`?

Reply:
From what I can tell from the TS'ing of client this should always be non undefined and non null. @vishalnarkhede might know more on this check? I also found it odd we are using break and not continue here?

Reply:
No clue about this check. It was there since biblical times :D

Reply:
We are going to come back to this when the TS is in.